pip_check_reqs/common.py

```diff
@@ -77,41 +77,32 @@ def _add_module(self, modname: str, lineno: int) -> None:
             return
         path = None
         progress = []
-        modpath = last_modpath = None
+        modpath = None
         for modname_part in modname.split("."):
-            try:
-                _, modpath, _ = imp.find_module(modname_part, path)
-            except ImportError:
-                # the component specified at this point is not importable
-                # (is just an attribute of the module)
-                # *or* it's not actually installed, so we don't care either
-                break
-
-            # success! we found *something*
-            progress.append(modname_part)
+            find_spec_result = importlib.util.find_spec(
+                name=modname_part,
+                package=path,
+            )
 
-            # we might have previously seen a useful path though...
-            if modpath is None:
-                # the `sys` module will hit this code path, and `os` will on
-                # 3.11+.
-                # Possibly others will, but I've not discovered them.
-                modpath = last_modpath
+            if find_spec_result is None:
+                # The component specified at this point is not installed.
                 break
 
-            # ... though it might not be a file, so not interesting to us
-            if not Path(modpath).is_dir():
-                break
+            modpath = find_spec_result.origin
 
-            path = [modpath]
-            last_modpath = modpath
+            # success! we found *something*
+            progress.append(modname_part)
 
         if modpath is None:
             # the module doesn't actually appear to exist on disk
             return
 
         modname = ".".join(progress)
         if modname not in self._modules:
-            self._modules[modname] = FoundModule(modname, modpath)
+            self._modules[modname] = FoundModule(
+                modname=modname,
+                filename=str(Path(modpath).parent),
```

Comment:
Why `.parent`? I think this might result in weirdness with libraries that install a single source file into `site-packages/`

Reply:
Just to match some existing behaviour, I believe.
Good spot! I found that this broke a real example in one of my projects, and I fixed it in https://github.com/r1chardj0n3s/pip-check-reqs/commit/344aff8e8c054ff87b5ae1662162397cdd49a3dc.